doc/design/context.md

```diff
@@ -0,0 +1,166 @@
+## Context Design
+
+
+A Net is executed by single or several threads. A Context is related to a thread and records necessary runtime resources.
+
+Context is defined as follows:
+
+```
+class Context {};
```

Comment:
The `Context` is just for unifying two class `CUDAContext` and `CPUContext`. And `RunContext` in mxnet is also for unifying two class `Stream<gpu>` and `Stream<cpu>`.